lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -45,3 +45,108 @@ function wp_theme_clean_theme_json_cached_data() {
 		WP_Theme_JSON_Resolver_Gutenberg::clean_cached_data();
 	}
 }
+
+/**
+ * Returns the stylesheet resulting of merging core, theme, and user data.
+ *
+ * @param array $types Types of styles to load. Optional.
+ *                     It accepts 'variables', 'styles', 'presets' as values.
+ *                     If empty, it'll load all for themes with theme.json support
+ *                     and only [ 'variables', 'presets' ] for themes without theme.json support.
+ *
+ * @return string Stylesheet.
+ */
+function gutenberg_get_global_stylesheet( $types = array() ) {
+	// Return cached value if it can be used and exists.
+	// It's cached by theme to make sure that theme switching clears the cache.
+	$can_use_cached = (
+		( empty( $types ) ) &&
+		( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
+		( ! defined( 'SCRIPT_DEBUG' ) || ! SCRIPT_DEBUG ) &&
+		( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST ) &&
+		! is_admin()
+	);
+	$cache_key      = 'gutenberg_get_global_stylesheet';
```

Comment:
Instead of making these values related to the current theme, why not just make the key the stylesheet name? That way, you do not have to invalidate on theme switching. The cache key would change.

Reply:
We can't assume the value cached can be reused, as the environment when it was saved may be different from when it is retrieved again (different plugin filterns, etc.).

Reply:
Sorry, can you explain this?

Reply:
> We can't assume the value cached can be reused

Yeah, it wouldn't work for example if the stylesheet of an inactive theme is cached and the theme gets updated with new styles. in that case, we should invalidate that cache either when an inactive theme is updated or after a theme switch to make sure that `gutenberg_get_global_stylesheet` doesn't provide a stale stylesheet.

But I think the proposed mechanism in this PR is simpler?